app/models/provider.rb

```diff
@@ -80,6 +81,19 @@ def restore!
     update!(archived_at: nil)
   end
 
+  def current_accreditations
+    accreditations.current.order_by_start_date
+  end
+
+  def has_current_accreditations?
+    current_accreditations.exists?
+  end
+
+  def sync_accreditation_status!
+    new_status = has_current_accreditations? ? "accredited" : "unaccredited"
```

Comment:
Not sure is this correct, kinda feels it needs to check for start and end covers current

Reply:
Oh right, this bit should really  be
```ruby
  def current_accreditations
    accreditations.current.order_by_start_date
  end
```

`order_by_start_date` is not really needed right?